src/basic/path-util.c

```diff
@@ -671,6 +671,9 @@ int find_executable_full(const char *name, bool use_path_envvar, char **ret_file
         if (!p)
                 p = DEFAULT_PATH;
 
+        if (exec_search_path)
+                p = exec_search_path;
+
```

Comment:
Changed to append to default PATH. However, which should we prefer in case the executable exists in both default PATH and ExecSearchPath? At the moment I believe it would go with the first that is found (which should be default).

Reply:
Changed to prepend to the default PATH. This way it will look into ExecSearchPath first and then if not found will look into the default paths.